src/main/java/com/stakater/nordmart/catalog/controller/CatalogController.java

```diff
@@ -61,4 +70,42 @@ public CatalogController(ProductRepository repository, MeterRegistry meterRegist
         productCount.set(productList.size());
         return productList;
     }
+
+    @ApiOperation(value = "Saves product.",
+            consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
+    @PostMapping(consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
+    public ProductDto saveProduct(@RequestBody ProductDto productDto) {
+        return catalogService.store(productDto);
+    }
+
+    @ApiOperation(value = "Updated product.",
+            consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
+    @PutMapping(consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
+    public ProductDto updateProduct(@RequestBody ProductDto productDto) {
+        return catalogService.update(productDto);
+    }
+
+    @ApiOperation(value = "Removes product by provided itemId. Returns 404 if product was not found.",
+            consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
+    @DeleteMapping(path = "/{itemId}")
+    public ResponseEntity<String> deleteProduct(@PathVariable(name = "itemId") String itemId) {
+        try {
+            catalogService.delete(itemId);
+            return new ResponseEntity("Successfully deleted item with id " + itemId, HttpStatus.OK);
+        } catch (NotFoundException e) {
+            return new ResponseEntity("Item with id " + itemId + " was not found.", HttpStatus.NOT_FOUND);
+        }
+    }
+
+    @ApiOperation(value = "Gets product by provided itemId. Returns 404 status if such product  was not found.",
+            consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
+    @GetMapping(path = "/{itemId}", produces = APPLICATION_JSON_VALUE)
+    public ResponseEntity getProduct(@PathVariable(name = "itemId") String itemId) {
+        Optional<ProductDto> productDto = catalogService.get(itemId);
+        if (productDto.isPresent()) {
+            return new ResponseEntity(productDto.get(), HttpStatus.OK);
+        } else {
+            return new ResponseEntity("Product with item id " + itemId + " not Found", HttpStatus.NOT_FOUND);
+        }
+    }
```

Comment:
As stated earlier, all of this is not handled in gateway or front-end. Or is this not supposed to be handled there ?

Reply:
Not part of the task I am working on actually. I have added them only as the initial implementation of CRUD feature. Mostly for the testing purposes.

Reply:
Okay great thanks, i wasn't aware of that.